components/service/glean/src/main/java/mozilla/components/service/glean/private/TimespanMetricType.kt

```diff
@@ -30,39 +32,58 @@ data class TimespanMetricType(
     private val logger = Logger("glean/TimespanMetricType")
 
     /**
-     * Start tracking time for the provided metric. This records an error if it’s
-     * already tracking time (i.e. start was already called with no corresponding
-     * [stopAndSum]): in that case the original start time will be preserved.
+     * Start tracking time for the provided metric and associated object. This
+     * records an error if it’s already tracking time (i.e. start was already
+     * called with no corresponding [stopAndSum]): in that case the original
+     * start time will be preserved.
+     *
+     * @param anyObject The object to associate with this timing.  This allows
+     * for concurrent timing of events associated with different objects to the
+     * same timespan metric.
      */
-    fun start() {
+    fun start(anyObject: Any) {
         if (!shouldRecord(logger)) {
             return
         }
 
-        TimespansStorageEngine.start(this)
+        TimingManager.start(this, anyObject)
```

Comment:
Please note that `TimingManager` is now shared across all the different metric types. Consider this example:

```
fun onPageLoaded(target: Event) {
  Pages.postLoadOperationTimespan.start(target)

  // some post load op happens

  // Accumulate the time
  Pages.onLoadDistribution.stopAndSum(target) // This will use postLoadOperationTimespan start time
  Pages.postLoadOperationTimespan.stopAndSum(target) // This will be broken?
}
```

I think the `TimingManager` should distinguish between the different metric types. This can either happen by having the manager be an instance within each relevant storage engine (thus making the storage engine depend on the timing manager) or by having the timing manager know about the "concept" of storage engines, having a structure like `Map[Engine, Map[Id, Time]]`. I'd personally go with the former, which gives us one less `object` to care about. But I have no strong opinion on this.

Reply:
I'm not sure this is an issue.  The current implementation uses `Map[MetricName, Map[Id, Time]]`.  Since metric names are globally unique (you can't have a timer and a timing distribution with the same name), this should be ok.  I'm not opposed to having separate `TimingManager` instances for each storage engine though as insurance against this.

Reply:
Oh, good point. Not needed!